misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -391,6 +405,15 @@ internal class RealTransacter private constructor(
         message.contains("restart transaction")
   }
 
+  /**
+   * "Transactions in TiKV encounter write conflicts". This can happen when optimistic transaction
+   * mode is on. Conflicts are detected during transaction commit
+   * https://docs.pingcap.com/tidb/dev/tidb-faq#error-9007-hy000-write-conflict
+   */
+  private fun isTidbWriteConflict(th: SQLException): Boolean {
```

Comment:
Does checking the mode when encountering this exception matter? Or is it only possible in optimistic mode?

Reply:
This is only possible in optimistic mode. In pessimistic mode the first write would acquire a lock and the second write would have to wait.